### PR TITLE
Feat(eos_cli_config_gen): Add vpn-ipv6 AF + vpn-ipv4 change

### PR DESCRIPTION
**Pull request by emilarista**
## Change Summary

Adds vpn-ipv6 address family.

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

Adds data model for vpn-ipv6:

```yaml
router_bgp:
  address_family_vpn_ipv6:
    domain_identifier: < string >
    peer_groups:
      < peer_group_name >:
        activate: < true | false >
    neighbor_default_encapsulation_mpls_next_hop_self:
      source_interface: < interface >
```

## How to test
Tested in lab + with molecule

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Comment by ClausHolbechArista** [comment]
please make this change non-breaking by still accepting the old `neighbor_default_encapsulation_mpls_next_hop_self`. Otherwise we will have to push this out to AVD v.4

**Comment by emilarista** [comment]
> please make this change non-breaking by still accepting the old `neighbor_default_encapsulation_mpls_next_hop_self`. Otherwise we will have to push this out to AVD v.4

Made changes to accept the old data model.

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

**Review by onurgashi — APPROVED** [review]

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-bgp.j2 [hunk 1]
```diff
@@ -34,7 +34,12 @@
 | Settings | Value |
 | -------- | ----- |
 {%             if router_bgp.peer_groups[peer_group].type is arista.avd.defined %}
-| Address Family | {{ router_bgp.peer_groups[peer_group].type }} |
+{%                 if router_bgp.peer_groups[peer_group].type is not string  %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-bgp.j2]: This `type` variable is only used in documentation and in some check in the validation role. Please don't introduce all this logic around it. Just use it as a string in your molecule. Ex. `type: "mpls"` or `type: "vpnv4, vpnv6"`

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-bgp.j2 (hunk 1 above)
Comment (emilarista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-bgp.j2]: Removed it as per suggestion.

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

**Pull request merged**